File: netflix-conductor/conductor-tools/src/test/java/com/stulsoft/netflix/conductor/tools/MetadataHelperTest.java
/*
 * Copyright (c) 2019. Yuriy Stul
 */

package com.stulsoft.netflix.conductor.tools;

import com.netflix.conductor.common.metadata.tasks.TaskDef;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.io.InputStream;
import java.util.Scanner;

import static org.junit.jupiter.api.Assertions.*;

/**
 * @author Yuriy Stul
 */
class MetadataHelperTest {

    @Test
    void taskDefFromJsonString() throws Exception {
        String text =readResource("taskDefinition.json");
        TaskDef taskDef = MetadataHelper.taskDefFromJsonString(text);
        assertNotNull(taskDef);
    }

    @Test
    void taskDefFromJsonString2() throws Exception {
        String text =readResource("taskDefinition2.json");
        TaskDef taskDef = MetadataHelper.taskDefFromJsonString(text);
        assertNotNull(taskDef);
    }

    private String readResource(String resourceName) throws IOException {
        InputStream is = this.getClass().getClassLoader().getResourceAsStream(resourceName);
        if (is == null)
            throw new IOException("Resource " + resourceName + " was not found");
        return new Scanner(is, "UTF-8").useDelimiter("\\A").next();
    }
}

File: netflix-conductor/conductor-tools/src/test/scala/com/stulsoft/netflix/conductor/tools/MetadataHelperTest.scala
/*
 * Copyright (c) 2019. Yuriy Stul
 */

package com.stulsoft.netflix.conductor.tools

import java.io.FileInputStream

import com.netflix.conductor.common.metadata.tasks.TaskDef
import org.scalatest.{FunSuite, Matchers}

import scala.util.{Failure, Success}

/**
 * @author Yuriy Stul
 */
class MetadataHelperTest extends FunSuite with Matchers {

  test("taskDefFromStream - one task") {
    val taskDefs = MetadataHelper.taskDefFromStream(new FileInputStream("src/test/resources/taskDefinition.json"))
    taskDefs.isSuccess shouldBe true
    taskDefs.get.length shouldBe 1
    taskDefs.get.head shouldBe a[TaskDef]
  }

  test("taskDefFromStream - multiple tasks") {
    val taskDefs = MetadataHelper.taskDefFromStream(new FileInputStream("src/test/resources/taskDefinitions.json"))
    taskDefs.isSuccess shouldBe true
    taskDefs.get.length shouldBe 2
    taskDefs.get.foreach(taskDef => taskDef shouldBe a[TaskDef])
  }

  test("taskDefFromStream - wrong JSON"){
    MetadataHelper.taskDefFromStream(new FileInputStream("build.sbt")).isFailure shouldBe true
  }
}
